interface/main/holidays/Holidays_Storage.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/**
+ * User: sharonco@matrix.co.il
+ * Date: 6/15/16
+ * This class contains all the interaction with the database that are used by the holidays/clinic closed events
+ */
+class Holidays_Storage{
+    const TABLE_NAME = "calendar_external";
+    const CALENDAR_CATEGORY_HOLIDAY = "6";
+    const CALENDAR_CATEGORY_CLOSED = "7";
+
+    /**
+     * This function selects ALL the holidays from the table calendar_external and returns an array
+     * @return array
+     */
+    public function get_holidays(){
+        $holidays= array();
+
+        $sql = sprintf('SELECT * FROM %s', self::TABLE_NAME);
+        $res=sqlStatement($sql);
+        while ($row = sqlFetchArray($res)) {
+            $holidays[] = $row;
+        }
+        return $holidays;
+    }
+
+    /**
+     * Selects  holidays/closed clinic events from the table events in a range of dates
+     * @param $star_date
+     * @param $end_date
+     * @return array [0=>"2016/06/16"]
+     */
+    public static function get_holidays_by_dates($start_date,$end_date){
+        $holidays= array();
+        $sql = sprintf(
```

Comment:
this is a good example why sprintf is not good. Need to instead use binding to deal with sql injection:
``````
$sql = (
  
            
  Write  Preview

this is a good example why sprintf is not good. Need to instead use binding to deal with sql injection:
``````
$sql = 'SELECT * FROM openemr_postcalendar_events WHERE (pc_catid = ? OR pc_catid = ?) AND pc_eventDate >= ? AND pc_eventDate <= ?';
$res=sqlStatement($sql, array(self::CALENDAR_CATEGORY_HOLIDAY,self::CALENDAR_CATEGORY_CLOSED,$start_date,$end_date) );